internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -955,6 +973,7 @@ func (t *Translator) translateBackendTrafficPolicyForGateway(
 			setIfNil(&r.TCPKeepalive, tf.TCPKeepalive)
 			setIfNil(&r.Timeout, tf.Timeout)
 			setIfNil(&r.DNS, tf.DNS)
+			r.StatName = buildRouteStatName(routeStatName, r.Metadata)
```

Comment:
shouldnt this be setIfNil ?

Reply:
the PR sets `setIfNil(&r.StatName, buildRouteStatName(routeStatName, r.Metadata))` in `applyTrafficFeatureToRoute`

Reply:
Yup, you are right. Done